qemu/tests/nested_test.py

```diff
@@ -0,0 +1,100 @@
+    """
+    Nested test:
+    1) Boot VM
+    2) Install ansible and related packages
+    2) Generate inventory file with L1 guest IP
+    3) Execute ansible command
+    4) Copy test results from L1 to host results directory
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    setup_bridge_sh = "/root/setup_bridge.sh"
+
+    def get_live_vms(env):
+        live_vms = []
+        for vm in env.get_all_vms():
+            if vm.is_alive():
+                live_vms.append(vm)
+        if not live_vms:
+            test.fail("No live VM.")
+        return live_vms
+
+    def generate_invent_file(env):
+        vms = get_live_vms(env)
+        invent_file = tempfile.mktemp()
+        ip_lst = list(map(lambda v: v.get_address(0), vms))
+        with open(invent_file, "w") as fd:
+            fd.writelines(ip_lst)
+        return invent_file
+
+    def copy_network_script(env):
+        deps_dir = virttest_data_dir.get_deps_dir()
+        file_name = os.path.basename(setup_bridge_sh)
+        br_file = os.path.join(deps_dir, file_name)
+        for vm in get_live_vms(env):
+            vm.copy_files_to(br_file, setup_bridge_sh)
+
+    sm = software_manager.SoftwareManager()
+    if not sm.check_installed("ansible"):
+        sm.install("ansible")
+
+    invent_file = generate_invent_file(env)
+
+    copy_network_script(env)
+
+    deps_dir = virttest_data_dir.get_deps_dir()
+
+    kar_cmd = "python3 ./ConfigTest.py "
+    test_type = params.get("test_type")
+    if test_type:
+        case_name = params.get("case_name")
+        if case_name:
+            kar_cmd += " --%s=%s " % (test_type, case_name)
+    l2_guest_name = params.get("l2_guest_name")
+    if l2_guest_name:
+        kar_cmd += " --guestname=%s" % l2_guest_name
+    clone = params.get("install_node")
+    if clone == "yes":
+        kar_cmd += " --clone=yes"
+    else:
+        kar_cmd += " --clone=no"
+
+    playbook_file = os.path.join(deps_dir, "playbook.yml")
+
+    ansible_cmd = "export ANSIBLE_SSH_ARGS=\"-C -o ControlMaster=auto -o ControlPersist=60s " \
+                  "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null\"; " \
+                  "ansible-playbook %s " \
+                  "--extra-vars '{\"command_line\": \"%s\", \"setup_br_sh\": \"%s\"}' " \
+                  "-i %s" \
+                  % (playbook_file, kar_cmd, setup_bridge_sh, invent_file)
+
+    logging.debug("ansible cmd: %s" % ansible_cmd)
+
+    try:
+        status, output = process.getstatusoutput(ansible_cmd, timeout=7200)
+        if status != 0:
+            test.fail(
+                "ansible_cmd failed, status: %s, output: %s" %
+                (status, output))
+    finally:
+        results_dir = os.path.dirname(test.logfile)
+        for vm in get_live_vms(env):
+            vm_results_dir = os.path.join(results_dir, vm.name)
+            os.makedirs(vm_results_dir)
```

Comment:
Suggest the target dir name to be `ansible_artifacts_$vm.name`

Reply:
Do copy logs task in playbook. Directory on host is like <host_log_dir>/<guest ip>/root/avocado/job-results/...